File: solutions/clojure/gigasecond/1/src/gigasecond.clj
(ns gigasecond)


(defn from [year month day]
 (let [date (java.time.LocalDate/of year month day)
       days-in-gigaseconds (quot 1000000000 (* 3600 24))
       final-date (.plusDays date days-in-gigaseconds)]
   ((juxt (memfn getYear) (memfn getMonthValue) (memfn getDayOfMonth)) final-date)))

File: solutions/java/gigasecond/1/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

public class Gigasecond {

    private LocalDateTime date;

    public Gigasecond(LocalDate moment) {
        this(moment.atTime(0, 0));
    }

    public Gigasecond(LocalDateTime moment) {
        date = moment.plusSeconds(1000000000);
    }

    public LocalDateTime getDateTime() {
        return date;
    }
}
